File: src/CSharp.Repository/Interfaces/IUsersRepository.cs
using CSharp.Forms.Data.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharp.Repository.Interfaces
{
    public interface IUsersRepository
    {

        Task<IEnumerable<User>> GetAll();
        Task<User> Get(string username);
        Task Update(User user);
        Task Delete(User user);
        Task Add(User user);
        Task<bool> UserExists(string username);
    }
}

File: src/VB.Repository/Interfaces/IUsersRepository.vb

Imports VB.Forms.Data.VB.Forms.Data

Namespace VB.Repository.Interfaces

    Public Interface IUsersRepository

        Function GetAll() As Task(Of IEnumerable(Of User))
        Function [Get](ByVal username As String) As Task(Of User)
        Function Update(ByVal user As User) As Task
        Function Delete(ByVal user As User) As Task
        Function Add(ByVal user As User) As Task
        Function UserExists(ByVal username As String) As Task(Of Boolean)

    End Interface


End Namespace


